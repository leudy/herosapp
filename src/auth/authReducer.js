import { types } from "../types/authTypes";

export const authReducer = (state = {}, action) => {


  
  switch (action.type) {
    case types.LOGIN: {
      const Data = { ...action.payload, logged: true };
      localStorage.setItem("user", JSON.stringify(Data));
      return Data;
    }
    case types.LOGOUT: {
      localStorage.removeItem("user");
      return { logged: false };
    }

    default:
      return state;
  }
};
