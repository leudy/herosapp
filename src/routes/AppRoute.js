import React from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { LoginScreen } from "../Login/LoginScreen";
import { ComicsRoute } from "./ComicsRoute";

export default function AppRoute() {
  return (
    <Router>
      <div>
        <Switch>
          <Route exact path="/login" component={LoginScreen}></Route>
          <Route path="/" component={ComicsRoute}></Route>
        </Switch>
      </div>
    </Router>
  );
}
