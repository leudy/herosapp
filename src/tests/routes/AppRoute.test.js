import React from "react";
import Adapter from "enzyme-adapter-react-16"
import {mount, configure} from "enzyme";    
import { MemoryRouter } from 'react-router-dom';
import { AuthContext } from "../../auth/AuthContext";
import AppRoute from "../../routes/AppRoute";

configure({adapter: new Adapter()});

describe('Prueba del <AppRoute .></AppRoute>', () => {
    // variabel fot test
    const contextValues  ={
        dispatch:jest.fn(),
        user:{
            logged:false
        }
    }
   test('deberia de retornar el login si no esta logeado ', () => {
       const wrapper = mount(
     <AuthContext.Provider value={contextValues}> 
         <AppRoute></AppRoute>
     </AuthContext.Provider>
       );
       console.log(wrapper.html())
       expect(  wrapper ).toMatchSnapshot();
   })
   
   test('should ruturn marvel catalogs', () => {
    const _contextValues  ={
        dispatch:jest.fn(),
        user:{
            logged:true,
            name:"Pedro Juan Varon"
             
        }
    }

    const wrapper = mount(
        <AuthContext.Provider value={_contextValues}> 
            <AppRoute></AppRoute>
        </AuthContext.Provider>
          );
          console.log(wrapper.html())
        //    expect(  wrapper.find("login").exists()),toBe("true");
   })
   
   

})
