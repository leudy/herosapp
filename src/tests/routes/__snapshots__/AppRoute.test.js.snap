// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del <AppRoute .></AppRoute> deberia de retornar el login si no esta logeado  1`] = `ReactWrapper {}`;
